### PR TITLE
Always use OpenSSL constants for Digest operations

### PR DESCRIPTION
**Pull request by dbussink**
### Summary

As also previously discussed in https://github.com/rails/rails/pull/40770#issuecomment-748347066, this moves the usage of Digest constants to always use the OpenSSL version of those Digest implementations.

### CODE REVIEW

**Comment by dbussink** [comment]
Does it make sense to also add a Rubocop rule that would instruct folks to use the `OpenSSL` version always?

**Comment by dbussink** [comment]
Also cc @rafaelfranca & @eileencodes since you helped out with previous changes in the linked PR for example.

**Review by eileencodes — COMMENTED** [review]
Can you add a changelog for this change?

**Comment by dbussink** [comment]
@eileencodes Which changelog should I use for this or should it be essentially added to all since this affects multiple gems?

**Comment by eileencodes** [comment]
Let's add it to all of the ones that changed. I was at first thinking Railties but since you can use Active Record and others outside of Rails, it makes sense to include it in all of the ones that changed.

**Comment by rafaelfranca** [comment]
> Does it make sense to also add a Rubocop rule that would instruct folks to use the `OpenSSL` version always?

I think it would help a lot.

**Comment by dbussink** [comment]
> Let's add it to all of the ones that changed. I was at first thinking Railties but since you can use Active Record and others outside of Rails, it makes sense to include it in all of the ones that changed.

Added!

> I think it would help a lot.

Is this something to handle separately or what is the process for this?

**Comment by dbussink** [comment]
@eileencodes I've updated this and fixed the merge conflicts as well. Is this worth merging then also without the linters yet?

**Comment by eileencodes** [comment]
I'll merge this and an issue to add a linter. Seems like a good first issue for someone looking to contribute. 😄

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
+*   OpenSSL constants are now used for Digest computations.
+
+    *Dirkjan Bussink*
+
 *   Relation#destroy_all perform its work in batches
 
+*   Prevent double saves in autosave of cyclic associations
```
Comment (abhaynikam) [inline activerecord/CHANGELOG.md:7]: I think some rebase conflict got added in changelog. The description below is for changes related to `Relation#destroy_all` in batches.

Comment (eileencodes) [reply activerecord/CHANGELOG.md:7]: Thanks! Fixed in ce7ff88a50c31d4ccea08a63ef8f698c5d25ba36

Comment (abhaynikam) [reply activerecord/CHANGELOG.md:7]: @eileencodes https://github.com/rails/rails/commit/ce7ff88a50c31d4ccea08a63ef8f698c5d25ba36 removed wrong changelog. 

We need to keep this line `* Relation#destroy_all perform its work in batches` and remove `* Prevent double saves in autosave of cyclic associations` 😊

**Pull request merged**